src/content/docs/essentials/database.mdx

```diff
@@ -0,0 +1,205 @@
+    Future<void> updateSpending(TableSpendings spending);
+}
+@DriftAccessor(tables: [Spendings])
+class SpendingsDao extends DatabaseAccessor<AppDatabase, Spendings>
+    with _$SpendingsDaoMixin implements SpendingsDataSource {
+        SpendingsDao(AppDatabase db) : super(db);
+...
+}
+```
+
+The repository will be used as a layer between the data from our table and the business logic of our application.
+
+```dart
+abstract class SpendingsRepository {
+    Future<List<Spendings>> getAllSpendings();
+    Future<void> insertSpending(Spendings spending);
+    Future<void> deleteSpending(Spendings spending);
+    Future<void> updateSpending(Spendings spending);
+}
+class SpendingsRepositoryImpl implements SpendingsRepository {
+    final SpendingsDataSource _spendingsDataSource;
+    SpendingsRepositoryImpl(this._spendingsDataSource);
+...
+}
+```
+
+### Writing data
+
+After we have created a table and data source for our expenses, we can write some data in it,
+
+```dart
+    await database.into(spendings).insert(
+      SpendingsCompanion.insert(
+        price: Value(3.9),
+        title: 'Coffee',
+        currency: Value('USD'),
+        date: DateTime.now(),
+        receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+      ),
+    );
+```
+
+delete data,
+
+```dart
+    await database.delete(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+    ).go();
+```
+
+or update data.
+
+```dart
+    await database.update(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+        set: SpendingsCompanion(
+            price: Value(2.9),
+            title: 'Coffee',
+            currency: Value('USD'),
+            date: DateTime.now(),
+            receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+        ),
+    ).go();
+```
+
+More detailed information about data writing operations, you can find  
+[here](https://drift.simonbinder.eu/docs/dart-api/writes/).
+
+### Reading data
+
+To read data, we need to create a query that will return the data we need.
+
+For example, we can retrieve all the data from the table
+
+```dart
+    final allSpendings = await database.select(spendings).get();
+```
+
+Or can filter the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..where((tbl) => tbl.id.equals(1));
+```
+
+sort the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..orderBy(
+            (tbl) => OrderingTerm(expression: tbl.price, mode: OrderingMode.desc),
+        );
+```
+
+or limit the data.
+
+```dart
+    final query = await database.select(spendings)
+        ..limit(10);
+```
+
+No one will stop us from combining everything together, but that's covered
+[here](https://drift.simonbinder.eu/docs/dart-api/select/).
+
+### Migrations
+
+It's often challenging to foresee precisely what data should be in the table right away. In the
+future, new fields or tables may emerge, and for such changes, a migration tool exists.
```

Comment:
Predicting initial data requirements for a table can be complex, as data needs often evolve, requiring new fields or tables. Migration tools are crucial for adapting the database structure to these changing requirements, ensuring flexibility and robustness.